pkg/reconciler/nscert/controller.go

```diff
@@ -57,8 +57,12 @@ func NewController(ctx context.Context, cmw configmap.Watcher) *controller.Impl
 	impl := namespacereconciler.NewImpl(ctx, c, func(impl *controller.Impl) controller.Options {
 		logger.Info("Setting up event handlers")
 		nsInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
-			FilterFunc: pkgreconciler.Not(pkgreconciler.LabelFilterFunc(
-				networking.DisableWildcardCertLabelKey, "true", false)),
+			FilterFunc: pkgreconciler.ChainFilterFuncs(
+				pkgreconciler.Not(pkgreconciler.LabelFilterFunc(
+					networking.DisableWildcardCertLabelKey, "true", false)),
+				pkgreconciler.Not(pkgreconciler.LabelFilterFunc(
+					networking.DisableInternalWildcardCertLabelKey, "true", false)),
```

Comment:
this will AND the predicates and you probably want OR?

Reply:
Yes, we need "OR".

And we probably should add the test case for it?

Reply:
We probably don't need filter function here at all as we still want to reconcile the namespace with the `disableWildcard = true` to clean up wildcard certificate (https://github.com/knative/serving/blob/master/pkg/reconciler/nscert/nscert.go#L80)

Reply:
After looking at this part again, we think the logic is actually correct:

```
Scenario 1:
    Expect: True

    DisableWildCard: False
    DisableInternalWildcard: False

    NotA: True
    NotB: True
    Chain: True

Scenario 2:
    Expect: False

    DisableWildCard: True
    DisableInternalWildcard: False

    NotA: False
    NotB: Unchecked
    Chain: False

Scenario 3:
    Expect: False

    DisableWildCard: False
    DisableInternalWildcard: True

    NotA: True
    NotB: False
    Chain: False

Scenario 4:
    Expect: False

    DisableWildCard: True
    DisableInternalWildcard: True

    NotA: False
    NotB: False
    Chain: False
```

Since we want the filter to only be true if there nothing is disabled.

Let me know if that's incorrect.

Reply:
Oops, didn't refresh page and see Zhimin's comment. That makes sense to me as the reconciler checks the labels and then deletes the certs.